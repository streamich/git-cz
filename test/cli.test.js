const pkg = require('../package.json');
const {runCLI} = require('./testUtils');

test('git-cz --help', async () => {
  const {getResult} = runCLI(['--help']);

  const result = await getResult();

  expect(result).toMatchSnapshot();
});

test('git-cz --version', async () => {
  const {getResult} = runCLI(['--version']);

  const result = await getResult();

  expect(result.trim()).toBe(pkg.version);
});

test('git-cz --non-interactive', async () => {
  const {getResult} = runCLI(['--non-interactive', '--dry-run']);

  const quote = process.platform === 'win32' ? '"' : '\'';
  const result = await getResult();

  expect(result).toBe(`\
Running in dry mode.
Will execute command:
git commit --file ${quote}.git/COMMIT_EDITMSG${quote}
Message:
chore: ðŸ¤– automated commit
`);
});
