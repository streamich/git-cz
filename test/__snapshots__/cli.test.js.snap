// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`git-cz --help 1`] = `
"
    Semantic emojified git commit, git-cz.

    Usage: git-cz [options]

    options:
        -h, --help          show usage information
        -v, --version       print version info and exit
        --disable-emoji     don't add emoji to commit title
        --format            custom formatting options for subject
        --non-interactive   run git-cz in non-interactive mode

    non-interactive mode options:
        --type              type of the commit, defaults to \\"chore\\"
        --subject           message of the commit, defaults to \\"automated commit\\"
        --scope             semantic commit scope
        --body              extended description of the commit
        --breaking          description of breaking changes, if any
        --issues            GitHub issues this commit closed, e.g \\"#123\\"
        --lerna             Lerna mono-repo packages this commit affects

"
`;
